datasets/tedlium/tedlium.py

```diff
@@ -0,0 +1,271 @@
+          available corpora: these files come from the TED-LIUM 2 release, but
+          have been modified to get a tokenization more relevant for English
+          language
+
+        Two corpus distributions:
+        - the legacy one, on which the dev and test datasets are the same as in
+          TED-LIUM 2 (and TED-LIUM 1).
+        - the 'speaker adaptation' one, especially designed for experiments on
+          speaker adaptation.
+        """,
+        citation="""\
+        @inproceedings{hernandez2018tedlium3,
+          title={TED-LIUM 3: twice as much data and corpus repartition for experiments on speaker adaptation},
+          author={Hernandez, Fran{\\c{c}}ois and Nguyen, Vincent and Ghannay, Sahar and Tomashenko, Natalia and Est{\\`e}ve, Yannick},
+          booktitle={International Conference on Speech and Computer},
+          pages={198--208},
+          year={2018},
+          organization={Springer}
+        }
+        """,
+        url="https://www.openslr.org/51/",
+        download_url="http://www.openslr.org/resources/51/TEDLIUM_release-3.tgz",
+        split_paths=[
+            (datasets.Split.VALIDATION, os.path.join("TEDLIUM_release-3", "legacy", "dev")),
+            (datasets.Split.TEST, os.path.join("TEDLIUM_release-3", "legacy", "test")),
+            # The legacy/train directory contains symlinks to "data",
+            # which are skipped by extraction (see above).
+            # Work around this by manually dereferencing the links here.
+            (datasets.Split.TRAIN, os.path.join("TEDLIUM_release-3", "data")),
+        ],
+    )
+
+    return [release1, release2, release3]
+
+
+class TedLium(datasets.GeneratorBasedBuilder):
+    """The TED-LIUM corpus is English-language TED talks, with transcriptions, sampled at 16kHz. It contains about 118 hours of speech."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = _make_builder_configs()
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "audio": datasets.features.Audio(sampling_rate=16_000),
+                "text": datasets.Value("string"),
+                "speaker_id": datasets.Value("string"),
+                "gender": datasets.features.ClassLabel(names=["unknown", "female", "male"]),
+                "file": datasets.Value("string"),
+                "id": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=self.config.description,
+            features=features,
+            supervised_keys=("audio", "text"),
+            homepage=self.config.url,
+            license=_LICENSE,
+            citation=self.config.citation,
+            task_templates=[AutomaticSpeechRecognition(audio_column="audio", transcription_column="text")],
+        )
+
+    def _split_generators(self, dl_manager):
+        data_dir = dl_manager.download_and_extract(self.config.download_url)
```

Comment:
To make the dataset streaming work with TAR archives, you have to use `dl_manager.iter_archive`, not `download_and_extract`, see librispeech_asr for example:

https://github.com/huggingface/datasets/blob/095d12ff7414df118f60e00cd6494299a881743a/datasets/librispeech_asr/librispeech_asr.py#L116-L130

Reply:
Hey @lhoestq, I making good progress! The issue I’m facing is that the data for all three splits (train, dev and test) is downloaded from one `.tar` file at http://www.openslr.org/resources/19/TEDLIUM_release2.tar.gz. The subsets are separated into sub-directories when the file is extracted (`TEDLIUM_release1/train`, `TEDLIUM_release1/dev`, `TEDLIUM_release1/test`). Thus, in defining `archive_path` as the `dl_manger.download`, the variable points to **all three subsets**, and the iterator `filepath` a generator object over the data for all three subsets: 
https://github.com/huggingface/datasets/blob/18ff32a52a857af4cfa997d9e192844478e11531/datasets/tedlium/tedlium.py#L194-L208
This means that in `_generate_examples`, the iterator `filepath` has to go through every example and check if it belongs to the correct split:
https://github.com/huggingface/datasets/blob/18ff32a52a857af4cfa997d9e192844478e11531/datasets/tedlium/tedlium.py#L246-L249
Does the `dl_manager` have functionality to produce an iterator for a specific subset of the data extracted from the `.tar` file?

Reply:
Feel free to host the data on the Hub and separate the splits in different files :)